### PR TITLE
Swagger updates to address docs and sdk issues 

### PR DESCRIPTION
**Pull request by finiteattractor**
Found few issues while generating docs and SDK. Addressed by these changes. All test pass and SDK generation and test cases passing as well. Changes:
- Fixed wrong enum definitions
- IncludedUpdateClassifications to use comma separated strings
- Fixed parameters not showing in docs
- removed discriminator property, was resulting in missing property in generated models
- First letter capital for better looking doc
- Cleaned common definition file
- fixed x-ms-enum without enum on same level error in code generation
- removed subscriptionid/resourcegroupname/azureautomationname parameters to simplify sdk (they are not in the client)
- Changed scheduleinfo to be a required property
- moved filter parameter for better sdk parameter ordering
- ran auto indent on files

### CODE REVIEW

**Comment by finiteattractor** [comment]
Note: This is a resubmit of original PR (https://github.com/Azure/azure-rest-api-specs/pull/2097). Been asked to resubmit from my own fork to address Travis issue with PRs originating from team forks.

**Review by marstr — COMMENTED** [review]
And just to confirm, this API Version is currently available, but in preview?

**Comment by dsgouda** [comment]
@marstr do you think you can review this since you have some context

File: specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/definitions.json [hunk 1]
```diff
@@ -1,631 +1,572 @@
 {
-    "swagger": "2.0",
-    "info": {
-      "title": "AutomationManagement",
-      "version": "2015-10-31"
+  "swagger": "2.0",
+  "info": {
+    "title": "AutomationManagement",
+    "version": "2015-10-31"
```
[the remark this answers is not in the document]
Comment (finiteattractor) [reply specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/definitions.json]: Nope, you caught a copy/paste mistake :) this should be matching the version of the folder. I fixed it. Thanks!!

[the remark this answers is not in the document]
Comment (finiteattractor) [reply specification/automation/resource-manager/Microsoft.Automation/2017-05-15-preview/definitions.json]: Correct, API is live and being used by Ibiza portal.

**Comment by ravbhatnagar** [comment]
@marstr - Only thing that popped from ARM point of view was that this has some breaking changes like making a property as required and enum changes. This if done in a new api-version is fine. Please confirm and looks good to merge other than that.

**Comment by finiteattractor** [comment]
@ravbhatnagar this is a new api version and it is required property so we should be ok here.

**Comment by marstr** [comment]
Adding the potential breaking changes label, but because this is in a new API Version, that's not a blocker.

**Review by marstr — APPROVED** [review]

**Pull request merged**